### ISSUE TITLE
Bug: tsr-declarations requires do not work with rootDir

### ISSUE THREAD

**Issue by andykais** [fabiandev/ts-runtime#21]
If I give a rootDir, and ask `tsr` to transpile a file that is not from the root directory, it assumes the `tsr-declaration.js` path incorrectly:
e.g.
```bash
tsr --tsConfig tsconfig.json src/types/assert.ts
```
will give the error
```bash
$ node lib/types/assert.js
Error: Cannot find module './tsr-declarations'
```
I think the core of this error is ts-runtime assumes I will always be generating runtime check files at the root of my repo. However, I just want runtime checks on a few functions in my repo that deal with user input.

tsconfig.json:
```json
{
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./lib/"
  },
  "include": ["src/**/*.ts"]
}
```

let me know if you want a reproducible repo, I can push one!

**Comment by fabiandev** [comment]
Thanks for reporting @andykais! If you have a reproducible repo ready that'd be great.

**Comment by andykais** [comment]
great! Here is a [sample repo](https://github.com/andykais/ts-runtime-test-rootdir-nested-file-transform)

**Comment by fabiandev** [comment]
Thanks! I will do my best to look into that soon.

**Comment by andykais** [comment]
thanks for working on this! Actually that fix almost works. I gave you a simplified test case, and it does pass, but I added a second to the same repo so you can see how the code errors out given a nested folder.

**Comment by fabiandev** [comment]
Can you summarize in just a few words what the remaining issue is. Thanks!

**Comment by andykais** [comment]
Yes I can. Apologies if this is confusing. The simplest way to describe the issue, is that when the tsconfig.json `outDir` is a nested folder (e.g. `./dist/nested`, not `./dist`), the tsr-declaration.js requires that tsr builds are incorrect.

In the sample repo I have described two test cases:
1. [test case](https://github.com/andykais/ts-runtime-test-rootdir-nested-file-transform/tree/master/tests/test-dist-folder) you fixed 4 days ago on `ts-runtime@next`
2. [test case](https://github.com/andykais/ts-runtime-test-rootdir-nested-file-transform/tree/master/tests/test-outdir-nested) the failing test case with a nested outDir

**Comment by fabiandev** [comment]
Sorry for getting back that late. I just tested again, and actually to me it seems that everything works as expected before I made the changes. Could it be that your setup was not quite right of *test-inline* with https://github.com/andykais/ts-runtime-test-rootdir-nested-file-transform/commit/f09d2d05f470054548049773030c661be362a96e and check again with `ts-runtime@^0.2.0` and your latest test cases?
